app/components/search.tsx

```diff
@@ -22,24 +25,35 @@ export default function Search({
   openQuestionTitles,
   onSelect,
   embedWithoutDetails,
+  queryFromUrl,
+  limitFromUrl,
+  removeQueryFromUrl,
 }: Props) {
-  const [showResults, setShowResults] = useState(false)
+  const [showResults, setShowResults] = useState(!!queryFromUrl)
   const [showMore, setShowMore] = useState(false)
   const searchInputRef = useRef('')
 
   const [urlSearchParams] = useSearchParams()
   const placeholder = urlSearchParams.get('placeholder') ?? 'Search for more questions here...'
 
-  const {search, arePendingSearches, results} = useSearch(onSiteAnswersRef)
+  const {search, arePendingSearches, results} = useSearch(onSiteAnswersRef, limitFromUrl)
 
-  const searchFn = (rawValue: string) => {
+  const searchFn = (rawValue: string, initialSearch = false) => {
     const value = rawValue.trim()
     if (value === searchInputRef.current) return
 
     searchInputRef.current = value
 
     search(value)
     logSearch(value)
+
+    if (queryFromUrl && !initialSearch) {
```

Comment:
I don't like this... How about wrapping the `  if (queryFromUrl && !searchInputRef.current) {` in a `useEffect` or something? That should allow ` initialSearch ` to be removed?

Reply:
yeah, I would have to wrap some functions in a callback which would cascade to more functions in other hooks, but that pattern sounds easier to understand => will do

Reply:
now updated + 1 more fix for browser back button when using `?q=...` initially, then running manual search, then going back